diff --git a/AwsEncryptionSDK/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/model/EncryptInput.java b/AwsEncryptionSDK/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/model/EncryptInput.java
index 67ac1304..7282539c 100644
--- a/AwsEncryptionSDK/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/model/EncryptInput.java
+++ b/AwsEncryptionSDK/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/encryptionsdk/model/EncryptInput.java
@@ -192,7 +192,7 @@ public class EncryptInput {
           "`frameLength` must be greater than or equal to 1"
         );
       }
-      if (this._frameLengthSet && this.frameLength() > 4294967296) {
+      if (this._frameLengthSet && this.frameLength() > 4294967296L) {
         throw new IllegalArgumentException(
           "`frameLength` must be less than or equal to 4294967296."
         );
diff --git a/mpl b/mpl
--- a/mpl
+++ b/mpl
@@ -1 +1 @@
-Subproject commit ea0fe508dca214b9c9bdaf0fc21122e83971ef1b
+Subproject commit ea0fe508dca214b9c9bdaf0fc21122e83971ef1b-dirty
